### ISSUE TITLE
simulator.initialise contains duplicate core entries

### ISSUE THREAD

**Issue by dougajmcdonald** [gareththegeek/corewar#38]
When I initialise the core, I sometime get duplicate addresses come back. 

This feels like it shouldn't be happening, let me know if I'm wrong! I logged all the instructions from a core initialisation here and got this:

```
{address: 7, opcode: 7, modifier: 1, aOperand: {…}, bOperand: {…}}
aOperand
:
{mode: 1, address: -2}
address
:
7
bOperand
:
{mode: 1, address: 0}
modifier
:
1
opcode
:
7
__proto__
:
Object
core.js:9 
{address: 7, opcode: 0, modifier: 4, aOperand: {…}, bOperand: {…}}
aOperand
:
{mode: 1, address: 0}
address
:
7
bOperand
:
{mode: 1, address: 0}
modifier
:
4
opcode
:
0
__proto__
:
Object
```

**Comment by gareththegeek** [comment]
This definitely would be an issue, each address in core can have only a single instruction!
Also, the value at each address is stored in an array and indexed by the address e.g. `this.instructions[address] = someinstruction` so if there is a duplicate it could put the entire array out of whack!

**Comment by gareththegeek** [comment]
Do you have a specific redcode which reproduces this issue so I can debug?

**Comment by gareththegeek** [comment]
I am struggling to reproduce this or see how it is possible just by looking at the code.  Need a concrete example so I can debug!

**Comment by dougajmcdonald** [comment]
I'll show you tomorrow, it's the dwarf that's doing it. It could be a manifestation of the 'init + step' thing we were talking about

**Comment by gareththegeek** [comment]
Should be fixed by #56

**Comment by gareththegeek** [comment]
Confirm fixed in corewar `v0.0.27`
